Ujaan/Lab-2/Part-3/flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description = "part 2 flake";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+    utils.url = "github:numtide/flake-utils";
+    hello_lib.url = "github:KingSyclone/nix-hello-lib";
+  };
+
+  outputs = { self, nixpkgs, utils, hello_lib }:
+  let
+    hello_app_overlay = final: prev: rec {
+      hello_app = final.callPackage ./default.nix {
+        hello_lib = hello_lib.packages."x86_64-linux".default;
```

Comment:
if you use the overlay from your `hello_lib` input you dont need to do this here. 

please dont do this as this locks you into one specific system for this package when evaluating your flake.